pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -522,20 +575,37 @@ func (ir *IngressReconciler) Reconcile(isvc *v1beta1.InferenceService) error {
 		} else {
 			hostPrefix = getHostPrefix(isvc, disableIstioVirtualHost, false)
 		}
-
+		host := network.GetServiceHostname(hostPrefix, isvc.Namespace)
+		scheme := "http"
 		isvc.Status.Address = &duckv1.Addressable{
 			URL: &apis.URL{
-				Host:   network.GetServiceHostname(hostPrefix, isvc.Namespace),
-				Scheme: "http",
+				Host:   host,
+				Scheme: scheme,
 			},
 		}
-		isvc.Status.SetCondition(v1beta1.IngressReady, &apis.Condition{
-			Type:   v1beta1.IngressReady,
-			Status: corev1.ConditionTrue,
-		})
-		return nil
+		if isIngressReady(isvc) {
+			// When the ingress has already been marked Ready for this generation,
+			// then it must have been successfully probed. This exception necessary for the case
+			// of global resyncs.
```

Comment:
maybe improve the comment here?
if the ingress is ready, it will not be probed, the opposite you mentioned.

maybe just do `if ! isIngressReady(isvc)` and probe as needed. wouldn't it be better?

Reply:
All the watched resources are periodically reconciled (by default for every 10 hours) by the controller irrespective of any change in the resource. In those scenarios it is unnecessary to probe the ingress, if it was already marked ready for the current generation (If it is marked true then it must be already successfully probed). I agree we can just use `if ! isIngressReady(isvc)` condition